presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -315,8 +315,8 @@ public PagesHashStrategy createPagesHashStrategy(List<Integer> joinChannels, Opt
     public PagesHashStrategy createPagesHashStrategy(List<Integer> joinChannels, Optional<Integer> hashChannel, Optional<JoinFilterFunction> joinFilterFunction)
     {
         try {
-            return joinCompiler.compilePagesHashStrategyFactory(types, joinChannels)
-                    .createPagesHashStrategy(ImmutableList.copyOf(channels), hashChannel);
+            return joinCompiler.compilePagesHashStrategyFactory(types, joinChannels, joinFilterFunction.map(f -> f.getClass()))
```

Comment:
also, instead of passing around an Optional<Class>, it seems like the only thing you need here is a boolean indicating whether a filter function exists?